### DISCUSSION TITLE
How to use a fallback in versionCompatibility

### DISCUSSION THREAD

**Discussion by aerialls** [renovatebot/renovate#31442] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab with the latest Renovate version

### Please tell us more about your question or problem

Hello,
I'm trying to generate a config to match multiple Docker images with several formats.

- `envoyproxy/envoy:v1.31.0`
- `envoyproxy/envoy:distroless-v1.31.0`

The second image is not handled directly by Renovate, as expected. I tried to use the `versionCompatibility` as documented with.

```
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["envoyproxy/envoy"],
      "versionCompatibility": "^(?<compatibility>.*)-(?<version>.*)$",
      "versioning": "semver"
    },
```

The issue is that now, the "standard" image `v1.31.0` is not recognized anymore.

Do I have a solution here to handle both syntax? One with the compatibility and also without the compatibility or do I need to handle both "versions" with the same regex but I will have a `versionCompatibility` without the  `<compatibility>` sometimes?

Thanks,

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
Try something like this:

```
    {
      "matchDatasources": ["docker"],
      "matchPackageNames": ["envoyproxy/envoy"],
      "versionCompatibility": "^((?<compatibility>.*)-)?(?<version>.*)$",
      "versioning": "semver"
    },
```

**Reply by aerialls** [reply]
Working perfectly, thanks!
